phalcon/validation/validator/uniqueness.zep

```diff
@@ -185,52 +186,97 @@ class Uniqueness extends CombinedFieldsValidator
 				}
 			}
 
-			let attribute = this->getColumnNameReal(record, this->getOption("attribute", singleField)),
-				except = this->getOption("except");
+			let isModel = record instanceof Model;
```

Comment:
You're in loop here. Let's get the instance type of the record immediately. So move this to the top.

Then, to avoid all the if/elseif confusion, let's separate the logic for the new collection parts. This way we can actually leave the existing Model code AS IS.
```
    if (record instanceof ...\Model) {
        $this->_isUniquenessModel(validation, field, record);
    }
    elseif (record instanceof ...\Collection) {
        $this->_isUniquenessCollection(validation, field, record);
    }
```

So move existing `isUniqueness()` to `_isUniquenessModel` and then create new one for Collection accordingly.